### PR TITLE
Modify rule S6249: fix Terraform code examples

### PR DESCRIPTION
**Pull request by GabinL21**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S6249/terraform/rule.adoc [hunk 1]
```diff
@@ -31,9 +33,9 @@ resource "aws_s3_bucket_policy" "mynoncompliantbucketpolicy" {
       {
         Sid       = "HTTPSOnly"
         Effect    = "Deny"
-        Principal = [
-          "arn:aws:iam::123456789123:root"
-        ] # secondary location: only one principal is forced to use https
+        Principal = {
```
Comment (GabinL21) [inline rules/S6249/terraform/rule.adoc:36]: The previous code example was "right" too, but we're not currently detecting it with our rule implementation. 

This change ensures that we're documenting a case that is working with our current implementation, while the implementation itself will be updated in other tickets.

Comment (jonas-wielage-sonarsource) [reply rules/S6249/terraform/rule.adoc:36]: Pasting the noncompliant example does not raise an issue in the current implementation (if I didn't do any mistake)

**Review by jonas-wielage-sonarsource — COMMENTED** [review]
Please change the PR title to reflect the RSPEC guidelines.
"Modify rule S6249: ..."

File: rules/S6249/terraform/rule.adoc (hunk 1 above)
Comment (jonas-wielage-sonarsource) [reply rules/S6249/terraform/rule.adoc:36]: Nevermind, was a mistake on my setup.

**Pull request merged**